nixos/modules/services/networking/sabnzbd/default.nix

```diff
@@ -0,0 +1,546 @@
+                      description = ''
+                        Whether the server supports TLS
+                      '';
+                      default = true;
+                    };
+                    ssl_verify = mkOption {
+                      type = enumFromAttrs {
+                        "strict" = 3;
+                        "allow injection" = 2;
+                        "none" = 0;
+                      };
+                      description = ''
+                        Level of TLS verification. Supported values:
+                        3, 'strict'          -- strict (normal) verification
+                        2, 'allow injection' -- allow locally injected certificates
+                        0, 'none'            -- no verification
+                      '';
+                      default = "strict";
+                    };
+                    expire_date = mkOption {
+                      type = types.nullOr types.str;
+                      description = ''
+                        If Notifications are enabled and an expiry date is
+                        set, warn 5 days before expiry. This setting
+                        does not automatically disable the server.
+                        Expected format: yyyy-mm-dd
+                      '';
+                      default = null;
+                    };
+                  };
+                }
+              );
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    warnings = lib.optional (cfg.configFile != null) ''
+      `sabnzbd.configFile` is deprecated, consider using `sabnzbd.settings` instead.
+      If you have values set in `sabnzbd.settings` set, they will be ignored.
+    '';
+
+    users.users = mkIf (cfg.user == "sabnzbd") {
+      sabnzbd = {
+        isSystemUser = true;
+        group = cfg.group;
+        description = "sabnzbd user";
+      };
+    };
+
+    users.groups = mkIf (cfg.group == "sabnzbd") {
+      sabnzbd = { };
+    };
+
+    systemd.services.sabnzbd =
+      let
+        files =
+          (lib.optional cfg.allowConfigWrite sabnzbdIniPath) ++ [ publicSettingsIni ] ++ cfg.secretFiles;
+        iniPathQuoted = lib.escapeShellArg sabnzbdIniPath;
+      in
+      {
+        description = "sabnzbd server";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
+          Type = "forking";
+          GuessMainPID = "no";
+          User = cfg.user;
+          Group = cfg.group;
+          StateDirectory = cfg.stateDir;
+          ExecStart = "${lib.getExe cfg.package} -d -f ${iniPathQuoted}";
+        };
+        preStart = ''
+          set -euo pipefail
+
+          ${lib.toShellVar "files" files}
+
+          ${lib.getExe (pkgs.python3.withPackages (py: [ py.configobj ]))} \
+            ${./config_merge.py} \
+            "''${files[@]}" | \
+          install -D -m ${if cfg.allowConfigWrite then "600" else "400"} \
```

Comment:
This should be really done with systemd-tmpfiles.

Reply:
Hi, I had some time to ponder this and I don't think it is going to work. Fundamentally, the file can't be generated at build time since secrets need to be merged into it. That means

1. we can't use tmpfiles to generate a symlink into the store, since the file isn't in the store
2. we can't use tmpfiles to ensure a file with r/o bits exists, since we need to write it

Discussion on Matrix suggested ReadOnlyPaths= as an alternative, but as far as I can see that runs into the same issue.